sdk/ml/azure-ai-ml/azure/ai/ml/_file_utils/file_utils.py

```diff
@@ -32,7 +33,7 @@ def traverse_up_path_and_find_file(path, file_name, directory_name=None, num_lev
     :return: Path to the file that we found, or an empty string if we couldn't find the file.
     :rtype: str
     """
-    current_path = Path(path)
+    current_path: Union[Path, str] = Path(path)
```

Comment:
If we are setting the value to `Path(path)`, this type should always be `Path`

Reply:
But I noticed that in line 49, current_path is assigned to str value:

![image](https://github.com/Azure/azure-sdk-for-python/assets/19650434/de3bea13-5e41-4502-a92f-08c6ea7c344f)

This is why I set current_path to Union[Path, str]

Reply:
Please let me know if this is acceptable, or I can use `type ignore` to avoid this mypy error. Thanks

Reply:
😮. I didn't expand my diff window enough. The type you have is correct.